src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/CSharpVirtualDocumentFactory.cs

```diff
@@ -101,18 +105,38 @@ public override bool TryCreateMultipleFor(ITextBuffer hostDocumentBuffer, [NotNu
         return virtualDocuments.Length > 0;
     }
 
-#pragma warning disable IDE0060 // Remove unused parameter
     private IEnumerable<ProjectKey> GetProjectKeys(Uri hostDocumentUri)
     {
-        // TODO: Actually return all of the projects that the host document is in
-        return new ProjectKey[] { default };
+        var projects = _projectSnapshotManagerAccessor.Instance.GetProjects();
+
+        var inAny = false;
+        var normalizedDocumentPath = FilePathNormalizer.Normalize(hostDocumentUri.GetAbsoluteOrUNCPath());
+        foreach (var projectSnapshot in projects)
+        {
+            // TODO: Can't call projectSnapshot.GetDocument as the paths are not normalized the same. Why?
```

Comment:
Are there specific work items tracking these to-dos?

Reply:
Oops, ignore this comment - I later saw you addressed the TODOs in other commits